tests/components/opentherm_gw/test_init.py

```diff
@@ -109,22 +99,10 @@ async def test_device_migration(
         name="Mock Gateway",
         manufacturer="Schelte Bron",
         model="OpenTherm Gateway",
-        sw_version=VERSION_OLD,
+        sw_version=VERSION_TEST,
     )
 
-    with (
-        patch(
-            "homeassistant.components.opentherm_gw.OpenThermGateway",
-            return_value=MagicMock(
-                connect=AsyncMock(return_value=MINIMAL_STATUS_UPD),
-                set_control_setpoint=AsyncMock(),
-                set_max_relative_mod=AsyncMock(),
-                disconnect=AsyncMock(),
-            ),
-        ),
-    ):
-        await setup.async_setup_component(hass, DOMAIN, {})
-
+    await setup.async_setup_component(hass, DOMAIN, {})
```

Comment:
```suggestion
    await hass.config_entries.async_setup(MOCK_CONFIG_ENTRY.entry_id)
```
Since we use config entries, we should avoid using `setupasync_setup_component`